hack/generator/pkg/codegen/testdata/ArmResource/Arm_test_embedded_resource_has_embedded_resource_inside_azure.golden

```diff
@@ -0,0 +1,712 @@
+
+// CreateEmptyArmValue returns an empty ARM value suitable for deserializing into
+func (bProperties *BProperties) CreateEmptyArmValue() interface{} {
+	return BPropertiesArm{}
+}
+
+// PopulateFromArm populates a Kubernetes CRD object from an Azure ARM object
+func (bProperties *BProperties) PopulateFromArm(owner genruntime.KnownResourceReference, armInput interface{}) error {
+	typedInput, ok := armInput.(BPropertiesArm)
+	if !ok {
+		return fmt.Errorf("unexpected type supplied for PopulateFromArm() function. Expected BPropertiesArm, got %T", armInput)
+	}
+	if typedInput.EnumField != nil {
+		enumFieldTyped := *typedInput.EnumField
+		bProperties.EnumField = &enumFieldTyped
+	}
+	var err error
+	if typedInput.RefField != nil {
+		var refField CResource
+		err = refField.PopulateFromArm(owner, *typedInput.RefField)
+		if err != nil {
+			return err
+		}
+		refFieldTyped := refField
+		bProperties.RefField = &refFieldTyped
+	}
+	return nil
+}
+
+// +kubebuilder:validation:Enum={"Qux"}
+type BPropertiesEnumField string
+
+const BPropertiesEnumFieldQux = BPropertiesEnumField("Qux")
+
+//Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/BResource
+type BResourceArm struct {
+	Id *string `json:"id,omitempty"`
+}
+
+//Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/CProperties
+type CProperties struct {
+	IntField    *int    `json:"intField,omitempty"`
+	StringField *string `json:"stringField,omitempty"`
+}
+
+var _ genruntime.ArmTransformer = &CProperties{}
+
+// ConvertToArm converts from a Kubernetes CRD object to an ARM object
+func (cProperties *CProperties) ConvertToArm(name string) (interface{}, error) {
+	if cProperties == nil {
+		return nil, nil
+	}
+	var result CPropertiesArm
+	if cProperties.IntField != nil {
+		intFieldTyped := *cProperties.IntField
+		result.IntField = &intFieldTyped
+	}
+	if cProperties.StringField != nil {
+		stringFieldTyped := *cProperties.StringField
+		result.StringField = &stringFieldTyped
+	}
+	return result, nil
+}
+
+// CreateEmptyArmValue returns an empty ARM value suitable for deserializing into
+func (cProperties *CProperties) CreateEmptyArmValue() interface{} {
+	return CPropertiesArm{}
+}
+
+// PopulateFromArm populates a Kubernetes CRD object from an Azure ARM object
+func (cProperties *CProperties) PopulateFromArm(owner genruntime.KnownResourceReference, armInput interface{}) error {
+	typedInput, ok := armInput.(CPropertiesArm)
+	if !ok {
+		return fmt.Errorf("unexpected type supplied for PopulateFromArm() function. Expected CPropertiesArm, got %T", armInput)
+	}
+	if typedInput.IntField != nil {
+		intFieldTyped := *typedInput.IntField
+		cProperties.IntField = &intFieldTyped
+	}
+	if typedInput.StringField != nil {
+		stringFieldTyped := *typedInput.StringField
+		cProperties.StringField = &stringFieldTyped
+	}
+	return nil
+}
+
+//Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/CResource
+type CResourceArm struct {
```

Comment:
yeah, agree that's weird - I think it's a consequence of the types being defined in "usage order" - the ARM type is "used by" the non-ARM type (because the non-arm type returns it from the ConvertToARM methods)